CONTRIBUTING.md

```diff
@@ -0,0 +1,140 @@
+1. After submitting your pull request, ensure that your contribution passes all status checks (e.g. passes all tests)
+
+Pull request review and approval is required by at least one project maintainer to merge.
+We will do our best to review the code additions in a timely fashion.
+Ensuring that you follow all steps above will increase our speed and ability to review.
+We will check for accuracy, style, code coverage, and scope.
+
+### Git commit messages
+
+For all commit messages, please use a short phrase that describes the specific change.
+For example, “Add feature to check normalization method string” is much preferred to “change code”.
+When appropriate, reference issues (via `#` plus number) .
+
+## Development
+
+### Overview
+
+The Software Gardening Almanac is primarily written in [Python](https://www.python.org/) through [Jupyter Book](https://jupyterbook.org/) with related environments managed by Python [Poetry](https://python-poetry.org/).
+[Node](https://nodejs.org) and [NPM](https://www.npmjs.com/) dependencies are also used to assist development activities.
+We use [GitHub actions](https://docs.github.com/en/actions) for [CI/CD](https://en.wikipedia.org/wiki/CI/CD) procedures such as automated tests.
+
+### Getting started
+
+To enable local development, perform the following steps.
+
+1. [Install Python](https://www.python.org/downloads/) (suggested: use [`pyenv`](https://github.com/pyenv/pyenv) for managing Python versions)
+1. [Install Poetry](https://python-poetry.org/docs/#installation)
+1. [Install Poetry environment](https://python-poetry.org/docs/basic-usage/#installing-dependencies): `poetry install`
+1. [Install Node](https://nodejs.org/en/download) (suggested: use [`nvm`](https://github.com/nvm-sh/nvm) for managing Node versions)
+1. [Install Node environment](https://docs.npmjs.com/cli/v9/commands/npm-install): `npm install`
+
+### Development Tasks
+
+Common development tasks are defined using [Poe the Poet](https://poethepoet.natn.io/), which helps simplify repeated use of commands.
+Poe the Poet is included as a Python Poetry dev group dependency and tasks may be used through the Poetry environment.
+Please see the [`pyproject.toml`](pyproject.toml) file's `[tool.poe.tasks]` table for a list of available tasks.
+
+For example:
+
+```shell
+# example of how Poe the Poet commands may be used.
+poetry run poe <task_name>
+
+# example of a task which runs jupyter-book build for the project
+poetry run poe build-book
+```
+
+### Linting
+
+Work added to this repo is automatically checked using [pre-commit](https://pre-commit.com/) via [GitHub Actions](https://docs.github.com/en/actions).
+Pre-commit can work alongside your local [git with git-hooks](https://pre-commit.com/index.html#3-install-the-git-hook-scripts)
+After [installing pre-commit](https://pre-commit.com/#installation) within your development environment, the following command also can perform the same checks:
+
+```sh
+% pre-commit run --all-files
```

Comment:
Is this in the PULL_REQUEST template? If not, consider adding - I think it could be a great addition to the checklist!

Reply:
Great idea, thank you, I'll get this added!